src/state/controls/Button.ts

```diff
@@ -71,4 +71,11 @@ export class Button extends Control<IButtonData> implements IButton {
     public giveInput(input: IButtonInput): Promise<void> {
         return this.sendInput(input);
     }
+
+    public update(changedData: Partial<IButtonData>): Promise<void> {
+        if (changedData.cooldown) {
```

Comment:
Here im trying to understand how to indicate that for this particular method that the cooldown should be a UTC whereas setCooldown takes a duration.

Could we perhaps use an extended interface here that has `duration` to allow for this behavior?

Trying to make this make sense for everyone.